src/Groups/matrices/transform_form.jl

```diff
@@ -0,0 +1,418 @@
+# TODO: in this file are used many methods TEMPORARILY defined in files matrix_manipulation.jl and stuff_field_gen.jl
+# once methods in those files will be deleted / replaced / modified, this file need to be modified too
+
+
+
+import AbstractAlgebra: Field
+
+export
+    iscongruent
+
+###############################################################################################################
+
+# Algorithm
+
+###############################################################################################################
+
+# The functions find_radical, block_anisotropic_elim and block_herm_elim are based on the article
+# James B. Wilson, Optimal Algorithms of Gram-Schmidt type, Linear Algebra and its Applications 438 (2013) 4573-4583
+
+# returns x,y such that ax^2+by^2 = c
+# at the moment, it simply search for x such that (c-ax^2)/b is a square
+# TODO: is there a faster way to find x and y?
+# TODO: it would be better if this is deterministic. This depends on gen(F) and issquare(F).
+function _solve_eqn(a::T, b::T, c::T) where T <: FieldElem
+   F = parent(a)  
+   ch = Int(characteristic(F))
+   dg = degree(F)
+   w = gen(F)
+   for i in collect(Iterators.product([0:ch-1 for j in 1:dg]...))
+      x = sum([w^(j-1)*i[j] for j in 1:dg])
+      s = (c - a*x^2)*b^-1
+#      vero, y = issquare(s)
+      if issquare(s) return x,square_root(s) end
+   end
+   return nothing, nothing
+end
+
+
+
+# if _is_symmetric, returns C,A,d where A*B*transpose(frobenius(A,e)) = C, C = block_matrix(2,2,[C,0,0,0]) and d = rank(C)
+# else returns C,A,d where B*A = C, C = [C 0] and d = rank(C)
+# Assumption: if _is_symmetric==true, then nr=nc always
+function find_radical(B::MatElem{T}, F::Field, nr::Int, nc::Int; e::Int=0, _is_symmetric::Bool=false) where T <: FieldElem
```

Comment:
```suggestion
function _find_radical(B::MatElem{T}, F::Field, nr::Int, nc::Int; e::Int=0, _is_symmetric::Bool=false) where T <: FieldElem
```
This should not be visible.

Reply:
Same for `block_anisotropic_elim` and `block_herm_elim`?

Reply:
Yes